invenio_rdm_records/services/files/tasks.py

```diff
@@ -0,0 +1,31 @@
+import io
+from flask import current_app
+from celery import shared_task
+from pypdf import PdfReader
+from invenio_db import db
+from invenio_rdm_records.proxies import current_rdm_records_service
+
+
+@shared_task(ignore_result=True)
+def extract_full_text(record_id, file_key):
+    """Extract full text."""
+    record = current_rdm_records_service.record_cls.pid.resolve(record_id)
+    status_file = record.media_files[f"{file_key}.txt"]
+    status_file.processor["status"] = "processing"
+    status_file.commit()
+    db.session.commit()
+    try:
+        with record.files[file_key].open_stream("rb") as fd:
+            reader = PdfReader(fd)
+            data = "\n".join(p.extract_text() for p in reader.pages).encode()
```

Comment:
We should clean up the extracted text. There are lots of blank space codes there. I'm not sure if the library allows for that.
I guess for a first iteration this is "good enough"